Android/Tapp/app/src/main/java/app/anish/com/tapp/dialogs/LinkedInWebViewLoginDialog.java

```diff
@@ -0,0 +1,294 @@
+                        LINKEDIN_SECRET_KEY
+                }
+                );
+    }
+
+    /**
+     * Method that generates the url for get the authorization token from the Service
+     * @return Url
+     */
+    private static String getAuthorizationUrl() {
+        return TextUtils.join("",
+                    new String [] {
+                            AUTHORIZATION_URL,QUESTION_MARK,RESPONSE_TYPE_PARAM,EQUALS,
+                            RESPONSE_TYPE_VALUE,AMPERSAND,CLIENT_ID_PARAM,EQUALS,LINKEDIN_API_KEY,
+                            AMPERSAND,STATE_PARAM,EQUALS,STATE, AMPERSAND, REDIRECT_URI_PARAM,
+                            EQUALS,REDIRECT_URI
+                    }
+                );
+    }
+
+    private void dismissDialogInError(String errorMsg) {
+        Log.e(LOG_TAG, errorMsg);
+        dismiss();
+        authListener.onAuthError(errorMsg);
+    }
+
+    private void dismissDialogInSuccess(AccessToken accessToken) {
+        dismiss();
+        authListener.onAuthSuccess(accessToken);
+    }
+
+
+    private class LinkedinWebViewClient extends WebViewClient {
+        @Override
+        public void onPageFinished(WebView view, String url) {
+            if (mProgressDialog != null && mProgressDialog.isShowing()) {
+                mProgressDialog.dismiss();
+            }
+        }
+
+
+        @Override
+        public boolean shouldOverrideUrlLoading(WebView webView, String authorizationUrl) {
+            Uri uri = Uri.parse(authorizationUrl);
+            if (uri.toString().startsWith(REDIRECT_URI)) {
+                String stateToken = uri.getQueryParameter(STATE_PARAM);
+                if (stateToken == null || !stateToken.equals(STATE)) {
+                    String msg = "State token doesn't match. Possible CSRF Attach";
+                    dismissDialogInError(msg);
+                    return true;
+                }
+
+                // If user doesn't allow authorization to our application,
+                // the authorizationToken will be null
+                String authorizationToken = uri.getQueryParameter(RESPONSE_TYPE_VALUE);
+                if (authorizationToken == null) {
+                    dismissDialogInError("The user didn't allow authorization");
+                    return true;
+                }
+
+                Log.i(LOG_TAG, "Auth token received " + authorizationToken);
+                String accessTokenUrl = getAccessTokenUrl(authorizationToken);
+
+                // Make the request in an Async Task
+                new PostRequestAsyncTask().execute(accessTokenUrl);
+            } else {
+                // default behaviour
+                String url = uri.toString();
+                Log.i(LOG_TAG, "Webview redirecting to : " + uri);
+                mWebView.loadUrl(url);
+            }
+
+            return true;
+        }
+    }
+
+    private class PostRequestAsyncTask extends AsyncTask<String, Void, Throwable> {
```

Comment:
The only way to make http requests in Android is to use Async Tasks. Do you suggest to completely remove the dependence between the calling activity and the dialog?